src/main/ApplicationImpl.cpp

```diff
@@ -285,6 +285,10 @@ ApplicationImpl::~ApplicationImpl()
     if (mWorkScheduler)
```

Comment:
this and next condition could be extracted - the same code is used here and in gracefulStop